crates/aptos-logger/src/macros.rs

```diff
@@ -60,9 +60,12 @@ macro_rules! log {
             concat!(file!(), ':', line!()),
         );
 
-        if METADATA.level() == $crate::Level::Error {
-            $crate::ERROR_LOG_COUNT.inc();
-        }
+        match METADATA.level() {
+            $crate::Level::Error => { $crate::ERROR_LOG_COUNT.inc() },
+            $crate::Level::Warn => { $crate::WARN_LOG_COUNT.inc() },
+            $crate::Level::Info => { $crate::INFO_LOG_COUNT.inc() },
```

Comment:
I am worried about perf because this counter is shared and it's in critical path of logging.

Reply:
metrics counters should be implemented in terms of atomic int increment, right? fast enough?

Reply:
if the _metric_ is slowing things down, the _logging_ is too much!

Reply:
This will lead to contention between totally unrelated threads, even if its atomic int. The CPU still needs to get a "lock" on the counter in some sense.

Reply:
We only check if a given log line is enabled down below, but we just increment the counter regardless.

Reply:
given recent too-much-logging issues, I think this is still important and acceptable. This metric will make it easier to catch too-much-logging at system test time, and then we shouldn't have too-much-logging in production. Do you have any alternatives or just "no, don't measure this"? I think atomic int counters are just about as cheap as we can get, unless we do fancier things with a thread-local counter and summed display, but I don't think we're doing that for any other metrics counters.

Reply:
I would rather collect these stats when we actually decide to print them, on the other side of the channel. That way it's not in the critical path.

Reply:
May be I wasn't clear earlier. This is still on the critical path. This macro puts the log event into a channel which gets polled by a task and processed. I was suggesting to count there, instead of here.

Reply:
`STRUCT_LOG_COUNT.inc();` is also currently on the call path before data gets to the queue. That's for all log events at any level.

The queue uses a try_send which may fail and drop the message. Counting things before that has value.

I still think atomic counters are pretty cheap and STRUCT_LOG_COUNT is okay and one more counter would be okay too.

Reply:
ok, moved all the counting to the thread on the other side of the logging channel